java/org/apache/catalina/util/TimeBucketCounter.java

```diff
@@ -97,23 +97,22 @@ public TimeBucketCounter(int bucketDuration, ScheduledExecutorService executorSe
      * Increments the counter for the passed identifier in the current time bucket and returns the new value.
      *
      * @param identifier an identifier for which we want to maintain count, e.g. IP Address
-     *
      * @return the count within the current time bucket
      */
     public final int increment(String identifier) {
-        String key = getCurrentBucketPrefix() + "-" + identifier;
+        String key = getBucketPrefix(System.currentTimeMillis()) + "-" + identifier;
         AtomicInteger ai = map.computeIfAbsent(key, v -> new AtomicInteger());
         return ai.incrementAndGet();
     }
 
     /**
-     * Calculates the current time bucket prefix by shifting bits for fast division, e.g. shift 16 bits is the same as
+     * Calculates the time bucket prefix by shifting bits for fast division, e.g. shift 16 bits is the same as
      * dividing by 65,536 which is about 1:05m.
-     *
+     * @param timestamp in millis
      * @return The current bucket prefix.
      */
-    public final int getCurrentBucketPrefix() {
-        return (int) (System.currentTimeMillis() >> this.numBits);
+    public final int getBucketPrefix(long timestamp) {
+        return (int) (timestamp >> this.numBits);
```

Comment:
Why make this change? It appears to increase code duplication with no obvious benefit.